### PR TITLE
Migrate BrandDesignUpdatePageViewModel onto LinearOnboardingOrchestrator

### PR DESCRIPTION
**Pull request by LukasPaczos**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1208671518894266/task/1215622484916139?focus=true
Tech Design URL (if applicable): https://app.asana.com/1/137249556945/project/481882893211075/task/1214772963645835

### Description
Stacked on https://github.com/duckduckgo/Android/pull/8840. This change moves the new-user linear onboarding flow onto that orchestrator, behind a kill switch, with no behaviour change.

It composes the new-user flow as a plan in `NewUserOnboardingPlanProvider` (the ordered steps, their preconditions, side effects, and the skip / quick-setup side plans), starts it from `LaunchViewModel` via `NewUserOnboardingPlanBootstrapper`, and routes `BrandDesignUpdatePageViewModel` through it. The page ViewModel picks one of two flow strategies at construction:

- `OrchestratorFlow` translates fragment callbacks into orchestrator events and renders the current step.
- `LegacyFlow` keeps the existing in-ViewModel state machine unchanged.

The `linearOnboardingOrchestrator` feature flag selects between them (enabled by default). Intra-dialog interactions (quick-setup rows, widgets, notification callbacks) stay in the ViewModel in both modes.

Pixels were verified to match across the base, skip, and skip-then-resume, quick-setup, and sync restore scenarios for three builds: the pre-change baseline, flag off, and flag on.

### Steps to test this PR

**Base flow via orchestrator**
- [x] Clean install
- [x] Launch the app
- [x] Go through linear onboarding
  - [x] Verify all steps are present and transition correctly
- [x] Reach the browser
  - [x] Verify the in-context CTAs show

**Skip->return flow via orchestrator**
- [x] Clean install
- [x] Launch the app
- [x] Click "I've been here before"
- [x] Click "Show Tutorial"
  - [x] Verify browser comparison step is shown
- [x] Go through linear onboarding
  - [x] Verify all steps are present and transition correctly.
- [x] Reach the browser
  - [x] Verify the in-context CTAs show

**Skip flow via orchestrator**
- [x] Clean install
- [x] Launch the app
- [x] Click "I've been here before"
- [x] Click "Start browsing"
- [x] Reach the browser
  - [x] Verify the native input box is visible and focused

**Restore sync flow via orchestrator**
- [x] Enable sync on the device
- [x] Uninstall the app (**don't clear data**)
- [x] Launch the app
  - [x] Verify you see the "Let's pick up where you left off" screen

**Quick setup flow via orchestrator**
Apply below diff:
```diff
diff --git a/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt b/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt
index b4ddba2126..a70907c829 100644
--- a/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt
+++ b/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt
@@ -45,6 +45,7 @@ class OnboardingQuickSetupExperimentManagerImpl @Inject constructor(
 ) : OnboardingQuickSetupExperimentManager {
 
     override suspend fun enroll(): QuickSetupExperimentVariant? = withContext(dispatcherProvider.io()) {
+        return@withContext QuickSetupExperimentVariant.TREATMENT
         toggles.onboardingQuickSetupExperimentJun3().enroll()
         when {
             toggles.onboardingQuickSetupExperimentJun3().isEnrolledAndEnabled(QuickSetupCohorts.TREATMENT) -> {
```
- [x] **Clear data** (to get rid of sync restore)
- [x] Launch the app
- [x] Click "I've been here before"
  - [x] Verify Quick setup page is visible
- [x] Change address bar position to bottom and start browsing
  - [x] Verify the bottom option is applied

Apply below diff:
```diff
diff --git a/app/src/main/java/com/duckduckgo/app/onboarding/LinearOnboardingOrchestratorFeature.kt b/app/src/main/java/com/duckduckgo/app/onboarding/LinearOnboardingOrchestratorFeature.kt
index 89716bbfda..1eb455e745 100644
--- a/app/src/main/java/com/duckduckgo/app/onboarding/LinearOnboardingOrchestratorFeature.kt
+++ b/app/src/main/java/com/duckduckgo/app/onboarding/LinearOnboardingOrchestratorFeature.kt
@@ -31,6 +31,6 @@ import com.duckduckgo.feature.toggles.api.Toggle.DefaultFeatureValue
     featureName = "linearOnboardingOrchestrator",
 )
 interface LinearOnboardingOrchestratorFeature {
-    @Toggle.DefaultValue(DefaultFeatureValue.TRUE)
+    @Toggle.DefaultValue(DefaultFeatureValue.FALSE)
     fun self(): Toggle
 }
```

**Base flow via VM**
- [x] Clean install
- [x] Launch the app
- [x] Go through linear onboarding
  - [x] Verify all steps are present and transition correctly
- [x] Reach the browser
  - [x] Verify the in-context CTAs show

**Skip->return flow via VM**
- [x] Clean install
- [x] Launch the app
- [x] Click "I've been here before"
- [x] Click "Show Tutorial"
  - [x] Verify browser comparison step is shown
- [x] Go through linear onboarding
  - [x] Verify all steps are present and transition correctly.
- [x] Reach the browser
  - [x] Verify the in-context CTAs show

**Skip flow via VM**
- [x] Clean install
- [x] Launch the app
- [x] Click "I've been here before"
- [x] Click "Start browsing"
- [x] Reach the browser
  - [x] Verify the native input box is visible and focused

**Restore sync flow via VM**
- [x] Enable sync on the device
- [x] Uninstall the app (**don't clear data**)
- [x] Launch the app
  - [x] Verify you see the "Let's pick up where you left off" screen

**Quick setup flow via VM**
Apply below diff:
```diff
diff --git a/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt b/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt
index b4ddba2126..a70907c829 100644
--- a/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt
+++ b/app/src/main/java/com/duckduckgo/app/onboardingquicksetup/OnboardingQuickSetupExperimentManager.kt
@@ -45,6 +45,7 @@ class OnboardingQuickSetupExperimentManagerImpl @Inject constructor(
 ) : OnboardingQuickSetupExperimentManager {
 
     override suspend fun enroll(): QuickSetupExperimentVariant? = withContext(dispatcherProvider.io()) {
+        return@withContext QuickSetupExperimentVariant.TREATMENT
         toggles.onboardingQuickSetupExperimentJun3().enroll()
         when {
             toggles.onboardingQuickSetupExperimentJun3().isEnrolledAndEnabled(QuickSetupCohorts.TREATMENT) -> {
```
- [x] **Clear data** (to get rid of sync restore)
- [x] Launch the app
- [x] Click "I've been here before"
  - [x] Verify Quick setup page is visible
- [x] Change address bar position to bottom and start browsing
  - [x] Verify the bottom option is applied

### UI changes
No changes expected.

---

> [!NOTE]
> **Medium Risk**
> First-run onboarding is a critical path with a large dual-path refactor; rollback relies on the remote flag, and orchestrator mode intentionally differs from legacy in at least one default-browser edge case covered by tests.
> 
> **Overview**
> **New-user onboarding** is wired to the shared **LinearOnboardingOrchestrator** behind remote toggle `linearOnboardingOrchestrator` (on by default). When enabled, `LaunchViewModel` starts the run via `NewUserOnboardingPlanBootstrapper` before opening `OnboardingActivity`; when disabled, behavior stays on the existing in-VM flow.
> 
> The PR adds **`NewUserOnboardingPlanProvider`**, which defines the full linear plan (intro, notifications, sync restore / reinstall / initial forks, comparison chart, default browser, address bar, input mode + preview, skip and quick-setup side plans), step events, pixels, and completion results (launch search/chat). **`BrandDesignUpdatePageViewModel`** picks **`OrchestratorFlow`** or **`LegacyFlow`** at construction: orchestrator mode maps UI callbacks to events and renders `resolveDialog()` output; legacy mode keeps the prior state machine unchanged.
> 
> **`OnboardingViewModel`** and **`OnboardingActivity`** defer terminal `AppStage` / skip writes and dev-skip navigation when the orchestrator is driving the run. **`onboarding-api` / `onboarding-impl`** are added as app dependencies; lint baselines record temporary `*-impl` imports in tests and orchestrator code.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5ce61f52a8c1ed733717429923f8f7d867ef0dab. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by LukasPaczos** [comment]
bugbot run

File: app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+            step.copy(
+                transition = { event ->
+                    if (event is NewUserOnboardingEvent.SkipNewUserOnboardingDevOptionClicked) AbortPlan else original(event)
+                },
+            )
+        }
+
+    private suspend fun resolveFirstDialog(ctx: NewUserOnboardingPlanContext): FirstDialog =
+        withContext(dispatchers.io()) {
+            val canRestore = withTimeoutOrNull(BLOCK_STORE_TIMEOUT_MS) { syncAutoRestore.canRestore() } ?: false
```
Comment (catalinradoiu) [inline app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt]: In the `BrandDesignUpdatePageViewModel` I see we are wrapping `syncAutoRestore.canRestore()` in a try/catch block. Should we do that here as well?

However when looking in other places of the app (besides the onboarding) where we are calling that method we don’t wrap it with any try/ catch so I’m wondering that’s the correct way to handle this.

File: app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+    private val onboardingQuickSetupExperimentManager: OnboardingQuickSetupExperimentManager,
+    private val quickSetupPixelSender: QuickSetupPixelSender,
+    private val inputScreenOnboardingWideEvent: InputScreenOnboardingWideEvent,
+    private val defaultBrowserDetector: DefaultBrowserDetector,
+    private val widgetCapabilities: WidgetCapabilities,
+    private val pixel: Pixel,
+    private val dispatchers: DispatcherProvider,
+) {
+
+    fun buildRootPlan(
```
Comment (catalinradoiu) [inline app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt:94]: From my understanding the user will see the splash screen until this method finished, right? I see it’s being called in the `LaunchViewModel`. It’s not a blocking comment, just trying to understand the current behavior.

Comment (LukasPaczos) [reply app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt:94]: That's right :+1: Users will see the splash screen until we transition to `OnboardingActivity`.

File: app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt (hunk 1 above)
Comment (LukasPaczos) [reply app/src/main/java/com/duckduckgo/app/onboarding/orchestrator/NewUserOnboardingPlanProvider.kt]: I'm not sure either, so I'll re-added the original check, probably was there for a reason.

**Review by catalinradoiu — APPROVED** [review]
Looks good and works as expected! 👍

File: app/src/main/java/com/duckduckgo/app/onboarding/ui/page/BrandDesignUpdatePageViewModel.kt [hunk 3]
```diff
@@ -636,6 +481,362 @@ class BrandDesignUpdatePageViewModel @Inject constructor(
...
+                NewUserOnboardingActivityDialog.Initial -> setCurrentDialog(INITIAL)
+                NewUserOnboardingActivityDialog.ComparisonChart -> setCurrentDialog(COMPARISON_CHART)
+                NewUserOnboardingActivityDialog.DefaultBrowserPrompt -> {
+                    val intent = defaultRoleBrowserDialog.createIntent(context)
+                    if (intent != null) {
+                        _commands.send(Command.ShowDefaultBrowserDialog(intent))
+                    } else {
+                        pixel.fire(AppPixelName.DEFAULT_BROWSER_DIALOG_NOT_SHOWN)
+                        orchestrator.onEvent(NewUserOnboardingEvent.DefaultBrowserPromptFinished(isDefaultBrowser = false))
+                    }
```
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply app/src/main/java/com/duckduckgo/app/onboarding/ui/page/BrandDesignUpdatePageViewModel.kt:811]: Not a bug. The reentry guard is owner-keyed, so it only throws when the **same `Job`** that already holds the lock re-locks it:

```kotlin
mutex.withLock(owner = currentCoroutineContext()[Job])
```

A different `Job` just waits on the lock as normal. The two coroutines here are different Jobs:

- The `onEvent` calls that drive state changes go through `emit()`, which wraps each one in `viewModelScope.launch { … }` (a fresh child Job).
- The state collector `observeOrchestratorState()` runs in its own Job via `launchIn(viewModelScope)`.

So when an `emit`-launched `onEvent` (Job A) holds the mutex, sets `_state.value`, and the collector (Job B) calls `orchestrator.onEvent` directly at L810, that nested call has `owner = Job B ≠ Job A`. The mutex makes Job B wait for the lock and then proceed once Job A releases — it does not throw.

The reentry warning in the orchestrator KDoc is about step/plan callbacks (`precondition` / `transition` / `onCompleted` / `result`), which do run synchronously on the lock holder's Job. `applyDialog` is a downstream StateFlow collector, not one of those callbacks. It also can't be holding the mutex when it reaches L810, because `withLock` releases before that `onEvent` returns.

I reproduced the exact interleaving in a test (advance to `DefaultBrowserPrompt` via an in-flight `onEvent`, then the collector fires the direct `onEvent` with a null intent) and the run completes with no `IllegalStateException`.
